### PR TITLE
Add an OCSP responder

### PR DESCRIPTION
**Pull request by bifurcation**
This PR adds a very basic OCSP responder to CFSSL.  It comes in few major parts:

1. A generic interface for an OCSP "source", which maps requests to responses
2. An instance of this interface that reads responses from a flat file 
3. An OCSP Responder struct that can respond for any source
4. A CLI command that assembles the whole thing together. (file -> source -> responder -> http)

### CODE REVIEW

**Comment by grittygrease** [comment]
Can you clean up the issues Travis is reporting with go vet?

File: cli/ocspserve/ocspserve.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"fmt"
+	"net/http"
+
+	"github.com/cloudflare/cfssl/cli"
+	"github.com/cloudflare/cfssl/log"
+	"github.com/cloudflare/cfssl/ocsp"
+)
+
+// Usage text of 'cfssl serve'
+var ocspServerUsageText = `cfssl serve -- set up a HTTP server handles OCSP requests from a file (see RFC 5019)
```
Comment (grittygrease) [inline cli/ocspserve/ocspserve.go]: Should change this from serve to ocspserve, same with line 16.

File: cli/cli.go [hunk 2]
```diff
@@ -66,6 +66,8 @@ type Config struct {
 	IP                string
 	Remote            string
 	Label             string
+	Responses         string
+	Path              string
```
Comment (lziest) [inline cli/cli.go]: I guess Path is not really necessary here, as we right now hard-code the endpoints. And yes we should later refactor serve and ocspserve with proper routing treatment.

File: cli/ocspserve/ocspserve.go (hunk 1 above)
Comment (kisom) [inline cli/ocspserve/ocspserve.go]: Maybe "server handles" -> "server that handles".

File: ocsp/responder.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+type Source interface {
+	Response(*ocsp.Request) ([]byte, bool)
+}
+
+// An InMemorySource is a map from base64(serialNumber) -> der(response)
+// It looks up a response purely based on serial number, without regard
+// to what issuer the request is asking for.
+type InMemorySource map[string][]byte
+
+func numberToKey(serialNumber *big.Int) string {
```
Comment (kisom) [inline ocsp/responder.go]: Why not just serialNumber.String()?

**Comment by grittygrease** [comment]
LGTM

**Comment by grittygrease** [comment]
@bifurcation Can you address the comments, this is almost ready to merge.

**Comment by kisom** [comment]
Needs rebase, then it should be good to go.

**Comment by grittygrease** [comment]
I rebased in #171

**Pull request closed without merging**